### PR TITLE
Full Window Audit

### PR DESCRIPTION
**Pull request by souricelle**
This PR has been closed.

### CODE REVIEW

**Comment by kevingranade** [comment]
Looks like some faction camp manifests have gone out of sync now, should be pretty easy to clear up, but I don't have time to do so at the moment.

**Comment by souricelle** [comment]
I'll do some testing and see if I can get it fixed, or at least document what exactly needs to get fixed. Thanks for the heads up.

File: data/json/furniture_and_terrain/terrain-windows.json [hunk 1]
```diff
@@ -737,24 +915,14 @@
   {
     "type": "terrain",
     "id": "t_metal_grate_window",
-    "name": "Window with metal grate",
-    "description": "Metal bars made into a grate for a window, Highly durability and will stand against any foe",
+    "name": "Barred reinforced boarded window",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/terrain-windows.json:918]: Please decapitalize all names.

**Comment by Night-Pryanik** [comment]
I don't like your removal of single-pane, double-pane etc windows. Even if they don't used anywhere in the mapgen, players could build them by themselves.

Since you're changing `id`s of the terrain, you need to apply migration for old ids into a new one. Otherwise there would be LOTS of errors on game load.

File: data/json/furniture_and_terrain/terrain-windows.json (hunk 1 above)
Comment (souricelle) [reply data/json/furniture_and_terrain/terrain-windows.json]: Done.

**Comment by souricelle** [comment]
> I don't like your removal of single-pane, double-pane etc windows. Even if they don't used anywhere in the mapgen, players could build them by themselves.
> 
> Since you're changing `id`s of the terrain, you need to apply migration for old ids into a new one. Otherwise there would be LOTS of errors on game load.

I really don't think anyone was using them at all. They were completely pointless and didn't even work properly, and we'd have had bug reports if people had made these faulty windows that couldn't be used properly. They're also redundant - is a normal house window single pane? double pane? Why is there a duplicate version of it? Additionally per Candlebury, a quadruple pane window isn't really something that would exist outside of serious custom jobs, and since there's no gameplay effect I can't think of a single argument for all that bloat.

I can use migration IDs, we could even migrate all the single/double/triple pane windows to just being normal house windows since they're functionally identical (except normal house windows actually work).

File: data/json/furniture_and_terrain/terrain-windows.json (hunk 1 above)
Comment (actual-nh) [reply data/json/furniture_and_terrain/terrain-windows.json]: > Please decapitalize all names.

Is this a style rule for terrain/furniture, or for items also?

**Comment by Night-Pryanik** [comment]
Unfortunately, this approach won't work. `migration` only works for items. You need to do the whole work in the source code.

**Comment by souricelle** [comment]
> Unfortunately, this approach won't work. `migration` only works for items. You need to do the whole work in the source code.

I wouldn't even know where to begin with that. I'd appreciate some more detailed direction but my C++ skills are very limited.

I should stress again that these windows were nonsensical, reduntant, nobody was using them, and they were completely improperly implemented. They have _got_ to go. If this is a matter of a couple of skippable error messages on startup for the four people who built them and are keeping their saves, I don't see a huge problem.

**Comment by Night-Pryanik** [comment]
Just look at the output of failed "general build matrix". There are already tons of errors. It's failed exactly because of that - the game can't find old IDs of the terrain you've changed.

**Comment by souricelle** [comment]
Can that be avoided by leaving the old IDs up, even if we're removing some?

**Comment by Night-Pryanik** [comment]
I hope this will help you
https://github.com/CleverRaven/Cataclysm-DDA/blob/fa0b7d6b03c7e50b215b1a87c3155904bf55e236/src/savegame_json.cpp#L4117-L4137

**Comment by souricelle** [comment]
Thanks! That is actually just what I needed.

**Comment by souricelle** [comment]
This isn't happening until after string freeze anyway, I had to fix a bunch of descriptions.

**Comment by souricelle** [comment]
Whew, that was a lot of work. If anyone has any further input let me know, but otherwise this one's done.

**Comment by Night-Pryanik** [comment]
@souricelle will this fix #39720?

**Comment by souricelle** [comment]
> @souricelle will this fix #39720?

Yes.

**Comment by souricelle** [comment]
> @souricelle will this fix #39720?

Yep!

**Comment by Night-Pryanik** [comment]
Why closing?

**Comment by souricelle** [comment]
> Why closing?

I elected to delete my fork and close my unmerged PRs following a personal dispute with Kevin. If my code is used, it is against my wishes.

**Comment by Night-Pryanik** [comment]
That's a shame.
Well, good luck to you!

**Comment by souricelle** [comment]
Sorry to disappoint, and thanks, you too.

**Comment by ZhilkinSerg** [comment]
> 
> 
> > Why closing?
> 
> I elected to delete my fork and close my unmerged PRs following a personal dispute with Kevin. If my code is used, it is being done against my wishes.
> 
> [He seems to think he has some moral imperative to use it,](https://i.imgur.com/g1CaYCP.png) but frankly I find that pretty gross.

Luckily that is not how licenses work.

**Pull request closed without merging**